src/db/sysdb_views.c

```diff
@@ -1688,7 +1687,7 @@ errno_t sysdb_add_group_member_overrides(struct sss_domain_info *domain,
             }
         }
 
-        val = talloc_strdup(obj, memberuid);
+        val = talloc_steal(obj, memberuid);
```

Comment:
Is it safe to steal `memberuid`?
In the case where `memberuid` was initially `NULL`, it's OK because it will be allocated by `add_domain_name()` attached to `tmp_ctx`. But when `memberuid` is not initially `NULL`, `memberuid` is assigned `orig_name` which was retrieved by `ldb_msg_find_attr_as_string()` and it returns an internal value (not a copy) which, I guess, is attached to the `msg`.

Reply:
> But when `memberuid` is not initially `NULL`, `memberuid` is assigned `orig_name` 

It's the other way around: "memberuid = orig_name" when override not found, i.e. "memberuid == NULL".

> which was retrieved by `ldb_msg_find_attr_as_string()` and it returns an internal value (not a copy) which, I guess, is attached to the `msg`.

Note that in all code paths `memberuid` is set to a pointer returned by `ldb_msg_find_attr_as_string()`, just different obj/attr is searched.

`talloc_steal()` will re-attach to a new context ('obj'). So when `talloc_free(tmp_ctx)` frees 'res_members'/'override_obj', this value should be left intact.

Do I miss something?

Reply:
OK. I had missed that `override_obj` was attached to `res_members`, which was in turn attached to `tmp_ctx`, which is freed at the end of the function.